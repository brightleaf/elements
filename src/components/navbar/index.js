import {
  NavBar,
  NavBarBrand,
  NavBarBurger,
  NavBarDivider,
  NavBarDropDown,
  NavBarEnd,
  NavBarItem,
  NavBarStart,
  NavBarMenu,
} from './navbar'

export {
  NavBar,
  NavBarBrand,
  NavBarBurger,
  NavBarDivider,
  NavBarDropDown,
  NavBarEnd,
  NavBarItem,
  NavBarStart,
  NavBarMenu,
}

export default NavBar
