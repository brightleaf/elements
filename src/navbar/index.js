import {
  NavBar,
  NavBarBrand,
  NavBarDivider,
  NavBarDropDown,
  NavBarEnd,
  NavBarItem,
  NavBarStart,
  NavBarMenu,
} from './navbar'

export {
  NavBar,
  NavBarBrand,
  NavBarDivider,
  NavBarDropDown,
  NavBarEnd,
  NavBarItem,
  NavBarStart,
  NavBarMenu,
}

export default NavBar
