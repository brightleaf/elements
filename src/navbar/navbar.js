import React, { useState } from 'react'
import classnames from 'classnames'
import { Colors } from '../modifiers'

export const NavBarItem = ({ children, className, isActive, ...props }) => {
  const kid = React.cloneElement(children, {
    // eslint-disable-next-line sonarjs/no-duplicate-string
    className: classnames('navbar-item', className, { 'is-active': isActive }),
    ...props,
  })
  return <>{kid}</>
}

export const NavBarBrand = ({
  href,
  src,
  className,
  target,
  onClick,
  width = 112,
  height = 28,
  title,
}) => {
  return (
    <div className="navbar-brand">
      <a className={classnames('navbar-item', className)} href={href}>
        {src && <img src={src} width={width} height={height} alt={title} />}
        {title}
      </a>

      <a
        role="button"
        className="navbar-burger burger"
        aria-label="menu"
        aria-expanded="false"
        data-target={target}
        onClick={onClick}
      >
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
        <span aria-hidden="true"></span>
      </a>
    </div>
  )
}

export const NavBarDivider = () => <hr className="navbar-divider" />

export const NavBarDropDown = ({ title, isRight, children, isBoxed }) => {
  const classes = classnames('navbar-item', 'has-dropdown', 'is-hoverable', {
    'is-right': isRight,
    'is-boxed': isBoxed,
  })

  const ddClasses = classnames('navbar-dropdown', {
    'is-boxed': isBoxed,
  })

  return (
    <div className={classes}>
      <a className="navbar-link">{title}</a>
      <div className={ddClasses}>{children}</div>
    </div>
  )
}
export const NavBarStart = ({ children }) => {
  return <div className="navbar-start">{children}</div>
}
export const NavBarEnd = ({ children }) => {
  return <div className="navbar-end">{children}</div>
}
export const NavBarMenu = ({ children, id, isActive }) => {
  return (
    <div
      id={id}
      className={classnames('navbar-menu', { 'is-active': isActive })}
    >
      {children}
    </div>
  )
}
export const NavBar = ({
  children,
  isTransparent,
  isFixedTop,
  isFixedBottom,
  isSpaced,
  isPrimary,
  isSuccess,
  isInfo,
  isWarning,
  isDanger,
  isLink,
  isLight,
  isWhite,
  isDark,
  isBlack,
  isText,
}) => {
  // is-transparent is-fixed-top  is-fixed-bottom
  const [isActive, setActive] = useState(false)

  const classes = classnames('navbar', {
    'is-transparent': isTransparent,
    'is-fixed-top': isFixedTop,
    'is-fixed-bottom': isFixedBottom,
    'is-spaced': isSpaced,
    ...Colors({
      isPrimary,
      isSuccess,
      isInfo,
      isWarning,
      isDanger,
      isLink,
      isLight,
      isWhite,
      isDark,
      isBlack,
      isText,
    }),
  })

  const kids = React.Children.map(children, (child, i) => {
    if (child.type === NavBarBrand) {
      return React.cloneElement(child, {
        onClick: e => {
          e.preventDefault()
          setActive(!isActive)
        },
      })
    }
    if (child.type === NavBarMenu) {
      return React.cloneElement(child, {
        isActive: isActive,
      })
    }
    return child
  })

  return (
    <nav className={classes} role="navigation" aria-label="main navigation">
      {kids}
    </nav>
  )
}
