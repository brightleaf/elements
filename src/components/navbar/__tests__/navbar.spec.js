import React from 'react'
import { render, cleanup, waitForDomChange } from '@testing-library/react'
import {
  toHaveClass,
  toHaveAttribute,
  toContainElement,
} from '@testing-library/jest-dom'
import { NavBar, NavBarItem } from '../navbar'

expect.extend({ toHaveClass, toHaveAttribute, toContainElement })

describe('The NavBar Component', () => {
  afterEach(cleanup)
  it('should render a nav', () => {
    const { container } = render(<NavBar>Test</NavBar>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('navbar')
  })
  it('should render a nav with classname is-primary when passed isPrimary', () => {
    const { container } = render(<NavBar isPrimary>Test</NavBar>)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('navbar')
    expect(container.firstChild).toHaveClass('is-primary')
  })

  it('should render a nav with classname is-fixed-top when passed isFixedTop', async () => {
    const { container } = render(<NavBar isFixedTop>Test</NavBar>)
    // await waitForDomChange({ container })
    expect(container.firstChild).toMatchSnapshot()
    expect(document.body).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('navbar')
    expect(container.firstChild).toHaveClass('is-fixed-top')
    expect(document.body).toHaveClass('has-navbar-fixed-top')
  })
  it('should render a nav with classname is-fixed-bottom when passed isFixedBottom', async () => {
    const { container } = render(<NavBar isFixedBottom>Test</NavBar>)
    expect(container.firstChild).toMatchSnapshot()
    expect(document.body).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('navbar')
    expect(container.firstChild).toHaveClass('is-fixed-bottom')
    expect(document.body).toHaveClass('has-navbar-fixed-bottom')
  })
  /*

  it('should render a div with the class number and the className passed in', () => {
    const { container } = render(<Number className="test" />)
    expect(container.firstChild).toMatchSnapshot()
    expect(container.firstChild).toHaveClass('number')
    expect(container.firstChild).toHaveClass('test')
  })
  it('should render with color and size modifiers', () => {
    const { container } = render(
      <Number hasBackgroundPrimary hasTextWeightBold hasTextWhite>
        23
      </Number>
    )
    expect(container).toMatchSnapshot()

    expect(container.firstChild).toHaveClass('number')
    expect(container.firstChild).toHaveClass('has-text-weight-bold')
    expect(container.firstChild).toHaveClass('has-text-white')
    expect(container.firstChild).toHaveClass('has-background-primary')
  })
  */
})
