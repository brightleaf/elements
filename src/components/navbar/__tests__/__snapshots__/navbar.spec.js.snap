// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The NavBar Component should render a nav 1`] = `
<nav
  aria-label="main navigation"
  class="navbar"
  role="navigation"
>
  Test
</nav>
`;

exports[`The NavBar Component should render a nav with classname is-fixed-bottom when passed isFixedBottom 1`] = `
<nav
  aria-label="main navigation"
  class="navbar is-fixed-bottom"
  role="navigation"
>
  Test
</nav>
`;

exports[`The NavBar Component should render a nav with classname is-fixed-bottom when passed isFixedBottom 2`] = `
<body
  class="has-navbar-fixed-bottom"
>
  <div>
    <nav
      aria-label="main navigation"
      class="navbar is-fixed-bottom"
      role="navigation"
    >
      Test
    </nav>
  </div>
</body>
`;

exports[`The NavBar Component should render a nav with classname is-fixed-top when passed isFixedTop 1`] = `
<nav
  aria-label="main navigation"
  class="navbar is-fixed-top"
  role="navigation"
>
  Test
</nav>
`;

exports[`The NavBar Component should render a nav with classname is-fixed-top when passed isFixedTop 2`] = `
<body
  class="has-navbar-fixed-top"
>
  <div>
    <nav
      aria-label="main navigation"
      class="navbar is-fixed-top"
      role="navigation"
    >
      Test
    </nav>
  </div>
</body>
`;

exports[`The NavBar Component should render a nav with classname is-primary when passed isPrimary 1`] = `
<nav
  aria-label="main navigation"
  class="navbar is-primary"
  role="navigation"
>
  Test
</nav>
`;
